### PR TITLE
CIRC-865 fix issue with 414 status during ending expired sessions

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- Use FindWithMultipleCqlIndexValues to get expired for items to avoid 414 Uri too long error;
- Cover this scenario by test;

Resolves: [CIRC-865](https://issues.folio.org/browse/CIRC-865)

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/test/java/api/loans/PatronActionSessionTests.java [hunk 1]
```diff
@@ -228,9 +234,29 @@ public void patronNoticesShouldBeSentWhenCheckInSessionIsEnded() {
...
+      .put(ACTION_TYPE, "Check-in");
+
+    PatronSessionRecord sessionRecord = PatronSessionRecord.from(representation);
+
+    assertThat(sessionRecord.getId().toString(), is(id));
+    assertThat(sessionRecord.getPatronId().toString(), is(patronId));
+    assertThat(sessionRecord.getLoanId().toString(), is(loanId));
+    assertThat(sessionRecord.getActionType(), is(PatronActionType.CHECK_IN));
+  }
+
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/PatronActionSessionTests.java:256]: Extract into a separate test class

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 2]
```diff
@@ -274,4 +279,63 @@ public void expiredSessionWithNonExistentUserShouldBeEnded() {
       .atMost(1, TimeUnit.SECONDS)
       .until(patronSessionRecordsClient::getAll, empty());
   }
+
+  @Test
+  public void shouldNotFailWithUriTooLargeErrorDuringEndingExpiredCheckOutSessions() {
+
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:285]: Remove blank line, please.

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 3]
```diff
@@ -274,4 +279,63 @@ public void expiredSessionWithNonExistentUserShouldBeEnded() {
...
+    expiredSessionProcessingClient.runRequestExpiredSessionsProcessing(204);
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(patronSessionRecordsClient::getAll, empty());
+  }
+
+  @Test
+  public void shouldNotFailWithUriTooLargeErrorDuringEndingExpiredCheckInSessions() {
+
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:299]: Remove blank line, please.

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 4]
```diff
@@ -274,4 +279,63 @@ public void expiredSessionWithNonExistentUserShouldBeEnded() {
...
+    expiredSessionProcessingClient.runRequestExpiredSessionsProcessing(204);
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(patronSessionRecordsClient::getAll, empty());
+  }
+
+  @Test
+  public void sessionsWithNotSpecifiedActionTypeShouldBeEnded() {
+
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:313]: Remove blank line, please.

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 5]
```diff
@@ -274,4 +279,63 @@ public void expiredSessionWithNonExistentUserShouldBeEnded() {
...
+    expiredSessionProcessingClient.runRequestExpiredSessionsProcessing(204);
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(patronSessionRecordsClient::getAll, empty());
+  }
+
+  private void createOneHundredPatronActionCheckOutSessions(String actionType) {
+    IntStream.range(0, 100).forEach(
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:326]: Please extract upper limit into a parameter

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 6]
```diff
@@ -274,4 +279,63 @@ public void expiredSessionWithNonExistentUserShouldBeEnded() {
...
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(patronSessionRecordsClient::getAll, empty());
+  }
+
+  @Test
+  public void sessionsWithNotSpecifiedActionTypeShouldBeEnded() {
+
+    createOneHundredPatronActionCheckOutSessions("");
+    List<JsonObject> sessions = patronSessionRecordsClient.getAll();
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:315]: Please extract this code into a private method

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionProperties.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package org.folio.circulation.domain.notice.session;
+
+public class PatronActionSessionProperties {
+
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionProperties.java:4]: Personally, I would remove this blank line as I don't think it increases the readability of this class

File: src/test/java/api/loans/PatronActionSessionTests.java (hunk 1 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/PatronActionSessionTests.java]: Done

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 2 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Done

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 3 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Done

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 4 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Done

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 6 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Done

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionProperties.java (hunk 7 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionProperties.java]: Done

**Pull request merged**